src/NuGet.CommandLine/Program.cs

```diff
@@ -49,39 +50,77 @@ public int Main(string[] args)
                     restore.OnExecute(async () =>
                         {
                             // Figure out the project directory
+                            IEnumerable<string> externalProjects = null;
+
                             PackageSpec project;
-                            var projectDirectory = projectFile.Value ?? Path.GetFullPath(".");
-                            if (string.Equals(PackageSpec.PackageSpecFileName, Path.GetFileName(projectDirectory), StringComparison.OrdinalIgnoreCase))
+                            var projectPath = Path.GetFullPath(projectFile.Value ?? ".");
+                            if (string.Equals(PackageSpec.PackageSpecFileName, Path.GetFileName(projectPath), StringComparison.OrdinalIgnoreCase))
                             {
                                 _log.LogVerbose($"Reading project file {projectFile.Value}");
-                                projectDirectory = Path.GetDirectoryName(Path.GetFullPath(projectDirectory));
-                                project = JsonPackageSpecReader.GetPackageSpec(File.ReadAllText(projectFile.Value), Path.GetFileName(projectDirectory), projectFile.Value);
+                                projectPath = Path.GetDirectoryName(projectPath);
+                                project = JsonPackageSpecReader.GetPackageSpec(File.ReadAllText(projectFile.Value), Path.GetFileName(projectPath), projectFile.Value);
+                            }
+                            else if (MsBuildUtility.IsMsBuildBasedProject(projectPath))
+                            {
+#if DNXCORE50
+                                throw new NotSupportedException();
```

Comment:
Seems like you could just have MSBuildUtility throw the NotSupportedException instead and remove one set of #ifdefs